src/shared/options.ts

```diff
@@ -3,53 +3,103 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as vscode from 'vscode';
 import { DocumentSelector } from 'vscode-languageserver-protocol';
-import { vscode, WorkspaceConfiguration } from '../vscodeAdapter';
 import * as path from 'path';
 
-export class Options {
-    constructor(
-        public commonOptions: CommonOptions,
-        public omnisharpOptions: OmnisharpServerOptions,
-        public languageServerOptions: LanguageServerOptions,
-        public razorOptions: RazorOptions
-    ) {}
+export interface CommonOptions {
```

Comment:
the diff probably won't make much sense side by side - I moved things around.  Essentially we now have 4 interface definitions: `CommonOptions`, `OmnisharpServerOptions`, `LanguageServerOptions`, and `RazorOptions`

Each of those has a class implementation that defines a getter for each option.  The getter reads the option value from vscode.

All of the class implementations are exported (`commonOptions`. `omnisharpOptions`, `languageServerOptions`, `razorOptions`, so to get an option, all a consumer needs to do is:
```
import { commonOptions } from '...options.ts'
const optionValue = commonOptions.dotnetPath
```

which will always return the most up to date value.

Reply:
I assume there are absolutely no perf implications from using a getter to read from the vscode options?

Reply:
if the value is needed extremely often in some loop, I'd advise getting it once outside of the loop.  But generally we don't use our options that way, we just access them on startup, or on a specific request which should not be impactful at all.